src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Services/BrokeredServicesUtilities.cs

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System.Threading.Tasks;
+using Microsoft;
+using Microsoft.ServiceHub.Framework;
+using NuGet.VisualStudio;
+
+using IBrokeredServiceContainer = Microsoft.VisualStudio.Shell.ServiceBroker.IBrokeredServiceContainer;
+using SVsBrokeredServiceContainer = Microsoft.VisualStudio.Shell.ServiceBroker.SVsBrokeredServiceContainer;
+
+namespace NuGet.PackageManagement.VisualStudio
+{
+    public static class BrokeredServicesUtilities
+    {
+        public static async ValueTask<IServiceBroker> GetRemoteServiceBrokerAsync()
+        {
+            await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
```

Comment:
Nothing in this method requires the main thread. You should be able to drop this. If `ServiceLocator` requires the main thread, per the 1st threading rule, it should get itself to the main thread since it's an async method.